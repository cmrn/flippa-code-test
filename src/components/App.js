import React from 'react';
import PropTypes from 'prop-types';
import './App.css';

const App = ({ counters, onAdd }) => (
  <div className="App">
    <header>
      Counter App
      <button onClick={() => onAdd('foo')}>Add Counter</button>
      <code>{JSON.stringify(counters)}</code>
    </header>
  </div>
);

App.propTypes = {
  counters: PropTypes.array.isRequired,
  onAdd: PropTypes.func.isRequired,
};

export default App;
