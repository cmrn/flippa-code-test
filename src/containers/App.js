import React from 'react';
import App from '../components/App';
import { view } from 'react-easy-state';
import counterStore from '../counterStore';

class AppContainer extends React.Component {
  constructor(props) {
    super(props);
    this.handleAdd = this.handleAdd.bind(this);
  }

  componentDidMount() {
    counterStore.load();
  }

  handleAdd(title) {
    counterStore.add(title);
  }

  render() {
    return <App {...this.props}
      counters={counterStore.counters} 
      onAdd={this.handleAdd}
    />;
  }
}

export default view(AppContainer);
