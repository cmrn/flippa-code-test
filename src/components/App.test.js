import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

const counters = [
  {id: "asdf", title: "bob", count: 1},
  {id: "qwer", title: "steve", count: 2}
];

describe("with no counters", () => {
  it('renders the title and AddCounter component', () => {
    const wrapper = shallow(<App counters={[]} onAdd={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("with some counters", () => {
  it('renders the title, AddCounter, and each Counter', () => {
    const wrapper = shallow(<App counters={counters} onAdd={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});