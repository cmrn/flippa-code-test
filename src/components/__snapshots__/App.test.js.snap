// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with no counters renders the title and AddCounter component 1`] = `
<div
  className="App"
>
  <h1>
    Counter App
  </h1>
  <AddCounter
    onAdd={[Function]}
  />
</div>
`;

exports[`with some counters renders the title, AddCounter, and each Counter 1`] = `
<div
  className="App"
>
  <h1>
    Counter App
  </h1>
  <AddCounter
    onAdd={[Function]}
  />
  <Counter
    count={1}
    key="asdf"
    onChange={[Function]}
    title="bob"
  />
  <Counter
    count={2}
    key="qwer"
    onChange={[Function]}
    title="steve"
  />
</div>
`;
